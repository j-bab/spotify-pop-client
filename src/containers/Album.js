import React from "react";
import Card from "react-bootstrap/esm/Card";

const Album = (props) => {
    const {artists, images, name, release_date, total_tracks} = props.item;
    let artistNames = artists.map(function (artist, index) {
        return <strong key={index}>{artist.name}</strong>;
    });

    return <Card>
        <Card.Img variant="top" src={images[0].url}/>
        <Card.Body>
            <Card.Title>{name}</Card.Title>
            <Card.Text>by {artistNames}
                <br/>
                {total_tracks} Tracks<br/>
                Released <em>{release_date}</em>
            </Card.Text>
        </Card.Body>
    </Card>;
};
export default Album;
