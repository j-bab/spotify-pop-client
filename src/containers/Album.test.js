import React from 'react';
import renderer from 'react-test-renderer';
import Album from './Album';


const item = {
    "album_type": "album",
    "artists": [
        {
            "external_urls": {
                "spotify": "https://open.spotify.com/artist/1dfeR4HaWDbWqFHLkxsg1d"
            },
            "href": "https://api.spotify.com/v1/artists/1dfeR4HaWDbWqFHLkxsg1d",
            "id": "1dfeR4HaWDbWqFHLkxsg1d",
            "name": "Queen",
            "type": "artist",
            "uri": "spotify:artist:1dfeR4HaWDbWqFHLkxsg1d"
        }
    ],
    "available_markets": [
        "CA",
        "US"
    ],
    "external_urls": {
        "spotify": "https://open.spotify.com/album/21HMAUrbbYSj9NiPPlGumy"
    },
    "href": "https://api.spotify.com/v1/albums/21HMAUrbbYSj9NiPPlGumy",
    "id": "21HMAUrbbYSj9NiPPlGumy",
    "images": [
        {
            "height": 640,
            "url": "https://i.scdn.co/image/ab67616d0000b273008b06ec71019afd70153889",
            "width": 640
        },
        {
            "height": 300,
            "url": "https://i.scdn.co/image/ab67616d00001e02008b06ec71019afd70153889",
            "width": 300
        },
        {
            "height": 64,
            "url": "https://i.scdn.co/image/ab67616d00004851008b06ec71019afd70153889",
            "width": 64
        }
    ],
    "name": "Jazz (Deluxe Remastered Version)",
    "release_date": "1978-11-10",
    "release_date_precision": "day",
    "total_tracks": 18,
    "type": "album",
    "uri": "spotify:album:21HMAUrbbYSj9NiPPlGumy"
};

test('Displays Album Correctly', () => {

    const component = renderer.create(
        <Album item={item}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});