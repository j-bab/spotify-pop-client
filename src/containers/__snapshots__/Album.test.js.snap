// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays Album Correctly 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top"
    src="https://i.scdn.co/image/ab67616d0000b273008b06ec71019afd70153889"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Jazz (Deluxe Remastered Version)
    </div>
    <p
      className="card-text"
    >
      by 
      <strong>
        Queen
      </strong>
      <br />
      18
       Tracks
      <br />
      Released 
      <em>
        1978-11-10
      </em>
    </p>
  </div>
</div>
`;
